### PR TITLE
Add aws_byte_buf_reset() function

### PR DESCRIPTION
**Pull request by danielsn**
*Description of changes:*
There are cases where the user wants to reuse the same buffer (we see this in some c-common tests, for example). Instead of either having to
1. poke inside the buf internals
1. call secure_zero, which has the added cost of a memset.

this gives the user an API function that does the right thing.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .cbmc-batch/jobs/aws_byte_buf_reset/aws_byte_buf_reset_harness.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/utils.h>
+
+void aws_byte_buf_reset_harness() {
+    struct aws_byte_buf buf;
+
+    __CPROVER_assume(is_bounded_byte_buf(&buf, MAX_BUFFER_SIZE));
+    ensure_byte_buf_has_allocated_buffer_member(&buf);
+    __CPROVER_assume(aws_byte_buf_is_valid(&buf));
+    assert(buf.capacity <= MAX_BUFFER_SIZE);
```
Comment (feliperodri) [inline .cbmc-batch/jobs/aws_byte_buf_reset/aws_byte_buf_reset_harness.c]: This is not necessary

File: .cbmc-batch/jobs/Makefile.aws_byte_buf [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (feliperodri) [inline .cbmc-batch/jobs/Makefile.aws_byte_buf]: Remove extra line

File: .cbmc-batch/jobs/aws_byte_buf_reset/aws_byte_buf_reset_harness.c (hunk 1 above)
Comment (danielsn) [reply .cbmc-batch/jobs/aws_byte_buf_reset/aws_byte_buf_reset_harness.c]: Removed.

File: .cbmc-batch/jobs/Makefile.aws_byte_buf (hunk 2 above)
Comment (danielsn) [reply .cbmc-batch/jobs/Makefile.aws_byte_buf]: Done

**Pull request merged**